lib/praxis-mapper/connection_factories/sequel.rb

```diff
@@ -0,0 +1,57 @@
+module Praxis::Mapper
+  module ConnectionFactories
+
+    class Sequel
+
+      def initialize(connection:nil, **opts)
+        raise ArgumentError, 'May not provide both a connection and opts' if connection && !opts.empty?
+
+        if connection
+          @connection = connection
+        else
+          @connection = ::Sequel.connect(**opts)
+        end
+
+        @timeout = @connection.pool.instance_variable_get(:@timeout)
+        @sleep_time = @connection.pool.instance_variable_get(:@sleep_time)
+
+        # connections that we created explicitly
+        @owned_connections = Hash.new
+      end
+
+      def checkout(connection_manager)
```

Comment:
Let's add some comments:
Follow the same Sequel flow when attempting to acquire a connection (i.e., max timeout, and polling interval)